docs/content/guides/dagster-pipes/aws-lambda.mdx

```diff
@@ -0,0 +1,288 @@
+
+- **Inside the body of the context manager (<PyObject object="open_dagster_pipes" module="dagster_pipes" />), retrieve a log and report an asset materialization.** These calls use the temporary communications channels established by <PyObject object="PipesDefaultContextLoader" module="dagster_pipes" /> and <PyObject object="PipesDefaultMessageWriter" module="dagster_pipes" />. To see the full range of what you can do with the <PyObject object="PipesContext" module="dagster_pipes" />, see the API docs or the general [Pipes documentation](/guides/dagster-pipes).
+
+At this point you can execute the rest of your AWS Lambda code as normal, invoking various <PyObject object="PipesContext" module="dagster_pipes" /> APIs as needed.
+
+### Step 1.3: Deploy the function
+
+When finished, click the **Deploy** button to update and deploy the function.
+
+---
+
+## Step 2: Create the Dagster objects
+
+In this step, you’ll create a Dagster asset that, when materialized, opens a Dagster pipes session and invokes the Lambda function you created in [Step 1](#step-1-create-a-lambda-function).
+
+### Step 2.1: Define the Dagster asset
+
+In your Dagster project, create a file named `dagster_lambda_pipes.py` and paste in the following code:
+
+```python
+# dagster_lambda_pipes.py
+
+from dagster import AssetExecutionContext, Definitions, asset
+from dagster_aws import PipesLambdaClient
+
+
+@asset
+def lambda_pipes_asset(context: AssetExecutionContext, lambda_pipes_client: PipesLambdaClient):
+  return lambda_pipes_client.run(
+      context: context,
+      client: "TODO",
+      function_name: "function_name_or_arn",
+      event: "TODO",
+  ).get_materialize_result()
+```
+
+Here’s what we did in this example:
+
+- Created an asset named `lambda_pipes_asset`
+
+- Provided <PyObject object="AssetExecutionContext" /> as the `context` argument to the asset. This object provides access to system APIs such as resources, config, and logging.
+
+- Specified a resource for the asset to use, <PyObject module="dagster_aws" object="PipesLambdaClient" />, which is a pre-built Dagster resource that allows you to quickly get Pipes working with AWS Lambda.
+
+  We also specified the following for the resource:
+
+  - `context` - The asset's `context` (<PyObject object="AssetExecutionContext" />) data
+  - `client` - A boto client that can authenticate to AWS
+  - `function_name` - The name or ARN of the function to invoke
+  - `event` - TODO: A JSON-serializable object to pass as input
```

Comment:
See previous comment about this - an example here would be helpful. I feel like the definition is a bit vague at the moment.

Reply:
I think you can either 
* just use an empty dict `{}` with a comment like `# this is where we would pass data our lambda needs to operate on` 
* flesh out the example to like pass `{'numbers': [1, 2, 3, 4, ...]}` and then have the `report_asset_materialization` return the sum as metadata 

In the other examples I believe we use the pipes `extras` section to pass data since those other targets have less convenient means to pass data. For lambda, the event payload has the same capabilities as the extras section so sort of takes precedent as the means to pass data since its more fundamental in the context of lambda.

Reply:
I ended up doing 
```event={"some_parameter_value": 1}``` in the `@asset` and `some_parameter_value = event['some_parameter_value']` in the lambda